### PR TITLE
Upgrade helm to v3.6.1

### PR DESCRIPTION
**Pull request by jrhouston**
### Description

Bumping the helm version as per #760 

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
...
```
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Review by dak1n1 — APPROVED** [review]
Tested locally. Looks good, thanks!

**Review by redeux — CHANGES_REQUESTED** [review]
Please hold this PR.  There is a regression in v3.6.1.  See: https://github.com/helm/helm/releases/tag/v3.6.2

**Review by dak1n1 — APPROVED** [review]
Dependencies look good, and it works in testing. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.3.0 (Unreleased)
+
+* Upgrade helm to 3.6.1
```
Comment (redeux) [inline CHANGELOG.md:3]: Needs to be updated as well

**Review by redeux — CHANGES_REQUESTED** [review]

**Review by redeux — APPROVED** [review]

**Pull request merged**